bundles/mapping/mapmodule/plugin/layers/LayersPlugin.ol3.js

```diff
@@ -239,25 +240,30 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapmodule.plugin.LayersPlugin',
         // change in UI also)
         // this is for optimization purposes
         var mapModule = this.getMapModule(),
-            mapLayers = mapModule.getOLMapLayers(layer.getId()),
-            mapLayer = mapLayers.length ? mapLayers[0] : null;
-
+            mapLayers = mapModule.getOLMapLayers(layer.getId());
+        if (!mapLayers || !mapLayers.length){
+            return;
+        }
         if(scaleOk && geometryMatch && layer.isVisible()) {
             // show non-baselayer if in scale, in geometry and layer visible
-            if (mapLayer && !mapLayer.getVisible()) {
-                mapLayer.setVisible(true);
-            } else {
-                return; //suppress event
+            if (!mapLayers[0].getVisible()) {
+                mapLayers.forEach(function (ol) {
+                    ol.setVisible(true);
+                });
+                this.notifyLayerVisibilityChanged(layer, scaleOk, geometryMatch);
             }
         } else {
             // otherwise hide non-baselayer
-            if (mapLayer && mapLayer.getVisible()) {
-                mapLayer.setVisible(false);
-            } else {
-                return; //suppress event
+            if (mapLayers[0].getVisible()) {
+                mapLayers.forEach(function (ol) {
+                    ol.setVisible(false);
+                });
+                this.notifyLayerVisibilityChanged(layer, scaleOk, geometryMatch);
             }
         }
-        var event = this._sandbox.getEventBuilder('MapLayerVisibilityChangedEvent')(layer, scaleOk, geometryMatch);
+    },
+    notifyLayerVisibilityChanged: function (layer, inScale, geometryMatch){
+        var event = this._sandbox.getEventBuilder('MapLayerVisibilityChangedEvent')(layer, inScale, geometryMatch);
```

Comment:
Change this._sandbox.getEventBuilder() -> Oskari.eventBuilder() to get rid of the deprecated warning.